### ISSUE TITLE
Add support for stripping optional peer dependencies

### ISSUE THREAD

**Issue by rtkt** [svanderburg/node2nix#316]
Hi.
Peer dependencies can be marked as optional in NPM's package.json since v7 [(proof)](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta) but looks like node2nix doesn't support stripping them.

Maybe I'll patch node2nix myself and open a pull request by the end of this week but I don't guarantee it.

**Comment by lilyinstarlight** [comment]
Yeah I neglected to implement that when updating node2nix last for npm v7. Node2nix is fairly unmaintained these days, but if you need it, feel free to open a PR to this repo and then open a PR to nixpkgs to pull in the patch (and ping me to review on nixpkgs)
